pkg/apis/core/types.go

```diff
@@ -4159,6 +4159,8 @@ const (
 	// HugePages request, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
 	// As burst is not supported for HugePages, we would only quota its request, and ignore the limit.
 	ResourceRequestsHugePagesPrefix = "requests.hugepages-"
+	// Default resource requests prefix
+	DefaultResourceRequestsPrefix = "requests."
```

Comment:
@derekwaynecarr Do we still need explicit types for first class resources or can we apply the logic this PR employs for first class resources too?

Reply:
i suspect we could consolidate logic now to say any compute resource (cpu, memory, etc.) could support requests.* or limits.* syntax is overcommittable.